src/crdt.rs

```diff
@@ -1262,11 +1262,20 @@ impl TestNode {
         let mut local_repair_addr = bind_addr;
         local_repair_addr.set_port(data.contact_info.tvu_window.port());
 
-        let transaction = UdpSocket::bind(local_transactions_addr).unwrap();
-        let gossip = UdpSocket::bind(local_gossip_addr).unwrap();
-        let replicate = UdpSocket::bind(local_replicate_addr).unwrap();
-        let repair = UdpSocket::bind(local_repair_addr).unwrap();
-        let requests = UdpSocket::bind(local_requests_addr).unwrap();
+        fn bind(addr: SocketAddr) -> UdpSocket {
+            match UdpSocket::bind(addr) {
+                Ok(socket) => socket,
+                Err(err) => {
+                    panic!("Failed to bind to {:?}: {:?}", addr, err);
+                }
+            }
+        };
+
+        let transaction = bind(local_transactions_addr);
```

Comment:
I *really* wanted to use a destructuring assignment here, something like the following pseudo-code, but I don't have the Rust skillz it seems:
```
let [
    transaction,
    gossip,
    replicate,
    repair,
    requests,
  ] = [
    local_transactions_addr,
    local_gossip_addr,
    local_replicate_addr,
    local_repair_addr,
    local_requests_addr,
  ].map(bind);
```

Reply:
`into_iter()`?

Reply:
This is the best I can do:
```rust
        let [transaction, gossip, replicate, repair, requests] = [
            local_transactions_addr,
            local_gossip_addr,
            local_replicate_addr,
            local_repair_addr,
            local_requests_addr,
        ].into_iter()
            .map(bind)
            .collect();
```
which fails to compile with:
```
error[E0282]: type annotations needed
    --> src/crdt.rs:1273:13
     |
1273 |         let [transaction, gossip, replicate, repair, requests] = [
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |             |
     |             cannot infer type for `_`
     |             consider giving the pattern a type
     |
     = note: type must be known at this point
```

Reply:
I don't see a way to .collect() into a tuple, which is what I really want I think